examples/vision/siamese.py

```diff
@@ -0,0 +1,209 @@
+
+"""
+# Similarity Network with Duplet Loss
```

Comment:
Suggestion (nit):

- @fchollet 's examples use sentence case for titles, so let's refactor as follows:

```suggestion
# Siamese network with a duplet loss
```

(I think we should use "Similarity learning" or "Siamese network" in the title, which is then expanded in the description.)

(Note that the original paper refers to the Siamese network as `"Siamese" network`)

Reply:
In the example, you use a contrastive loss, while "duplet" refers to "duplet network". This may be be confusing for the reader, since in the title and the description, you mention the "duplet loss".

Reply:
Thanks @8bitmp3 for pointing out this.
I was assuming that Contrastive Loss is a generic loss function which just contrasts the positive and negative labeled images, this contrast can be done via two variants (Duplet Loss and Triplet Loss). Duplet Loss is Contrastive Loss which make use of a pair of input images from Duplet Network (Two Networks each with its own input), Triplet Loss is Contrastive Loss which make use of input images from Triplet Network (Three Networks). 

Please jump to 3.2 in [Learning Similarity Metric](https://archives.ismir.net/ismir2019/paper/000057.pdf)

Guide me If I am mistaken. Thanks